### DISCUSSION TITLE
How to run `yarn install` on CI without `--immutable`

### DISCUSSION THREAD

**Discussion by segrey** [yarnpkg/berry#3486] [Q&A] [answered]
We run `yarn install` on CI, and it fails with https://yarnpkg.com/advanced/error-codes#yn0028---frozen_lockfile_exception.
We'd rather avoid committing `yarn.lock`, because tests are running against latest yarn berry version and this might imply updating `yarn.lock`. Actually, we don't need exact installed package versions, so no real need in `yarn.lock`.

The tests are running on TeamCity CI. As I can see, `--immutable` is implied if `CI` or `TEAMCITY_VERSION` environment variables are set. Unfortunately, it's not enough. Could you please point me out to code that calculates `--immutable` default value?

**Comment by merceyz** [comment, marked as answer]
There are a multiple options here so I'll list them

- You can pass the `--no-immutable` flag to turn it off for the install command
```sh
yarn install --no-immutable
```

- Following the documentation for the configuration (https://yarnpkg.com/configuration/yarnrc) you can use environment variables to turn it off regardless of the command
```sh
YARN_ENABLE_IMMUTABLE_INSTALLS=false yarn
```
- Or you can disable it in the `.yarnrc.yml` config
```sh
yarn config set enableImmutableInstalls false
```

> Could you please point me out to code that calculates --immutable default value?

It's based on `ci-info` detecting a CI here https://github.com/yarnpkg/berry/blob/dd043a8ab1618c0b02e28b2b40743bd336d63437/packages/plugin-essentials/sources/index.ts#L91

**Reply by wajeehmisbahkhan** [reply]
I tried adding `YARN_ENABLE_IMMUTABLE_INSTALLS=false` to my `.env` file when deploying firebase functions but it doesn't seem to override the `--immutable` option (which I assume is being passed as I keep receiving the same lockfile modification forbidden error). Has anyone had luck implementing this solution when deploying firebase functions?

**Comment by segrey** [comment]
Wow, that was fast! Thank you @merceyz!

**Comment by say8425** [comment]
```sh
yarn install --no-immutable
```

I'm just curious, there's no `--no-immutable` option in the [yarn install docs](https://yarnpkg.com/cli/install). `--no-immutable` is still works in yarn berry?

**Reply by cmoulliard** [reply]
Yes - it still works with berry. I'm using it with yarn >= 4.x and github action flow
